### PR TITLE
Feature/ecu battery module send notification & use logger in class

### PR DESCRIPTION
**Pull request by nagyyalexandru**
- removed unnecessary thread in battery class
- response with Up notification will be sent to MCU from 'receive frames' module (have talked to Miruna, and added some additional modifications in PR #76)
- added logger functionality in battery class

### CODE REVIEW

File: src/ecu_simulation/BatteryModule/src/BatteryModule.cpp [hunk 1]
```diff
@@ -34,44 +34,33 @@ BatteryModule::BatteryModule(int _interfaceNumber, int _moduleId) : moduleId(_mo
...
-    GenerateFrames g1 = GenerateFrames(canInterface.getSocketFd());
+    GenerateFrames notifyFrame = GenerateFrames(canInterface.getSocketFd());
 
     /* Create a vector of uint8_t (bytes) containing the data to be sent */
     std::vector<uint8_t> data = {0x0, 0xff, 0x11, 0x3};
 
     /* Send the CAN frame with ID 0x22110 and the data vector */
-    g1.sendFrame(0x22110, data);
-}
+    notifyFrame.sendFrame(0x22110, data);
```
Comment (Miruna-Maria) [inline src/ecu_simulation/BatteryModule/src/BatteryModule.cpp:62]: id is 0x2210

File: src/ecu_simulation/BatteryModule/src/BatteryModule.cpp [hunk 2]
```diff
@@ -140,19 +129,23 @@ void BatteryModule::fetchBatteryData()
         this->energy = energy;
         this->voltage = voltage;
         this->percentage = percentage;
-#ifdef BATTERY_MODULE_DEBUG
-        std::cout << "Fetched Data - Energy: " << energy << " Wh, Voltage: " << voltage << " V, Percentage: " << percentage << "%, State: " << state << std::endl;
-#endif
+
+        // LOG_INFO(batteryModuleLogger.GET_LOGGER(), "Battery Data : Energy: " + std::to_string(energy) + " Wh, Voltage: "
+        //     + std::to_string(voltage) + " V, Percentage: " + std::to_string(percentage) + " %, State: " + state);
```
Comment (Miruna-Maria) [inline src/ecu_simulation/BatteryModule/src/BatteryModule.cpp:134]: comments with /* */

Comment (nagyyalexandru) [reply src/ecu_simulation/BatteryModule/src/BatteryModule.cpp]: that lines were not necessary, i deleted them

File: src/ecu_simulation/BatteryModule/src/BatteryModule.cpp (hunk 1 above)
Comment (nagyyalexandru) [reply src/ecu_simulation/BatteryModule/src/BatteryModule.cpp:62]: fixed

**Review by GeorgeCimpoeru — APPROVED** [review]

**Comment by GeorgeCimpoeru** [comment]
Solve the conflicts

**Comment by nagyyalexandru** [comment]
solved conflicts

File: src/ecu_simulation/BatteryModule/src/BatteryModule.cpp (hunk 1 above)
Comment (Mihnea9901) [reply src/ecu_simulation/BatteryModule/src/BatteryModule.cpp:62]: maybe update the id to 0x1110 as we talked in the daily??

Comment (nagyyalexandru) [reply src/ecu_simulation/BatteryModule/src/BatteryModule.cpp:62]: done

**Review by Mihnea9901 — APPROVED** [review]

**Review by Miruna-Maria — APPROVED** [review]

**Pull request merged**